realm/src/main/java/io/realm/Realm.java

```diff
@@ -191,6 +195,11 @@ public void close() {
             realmsCache.get().remove(id);
             sharedGroup.close();
             sharedGroup = null;
+            AtomicInteger counter = openRealms.get(id);
+            if (counter == null) {
+                throw new AssertionError("The counter should not be zero");
```

Comment:
I assume this assertion should not be possible, it's basically unneeded code as the decrementAndGet would fail with nullException anyway. So is there much value in having this?